app/src/main/java/gifapp/jsm/myapp/app/interview/jsmtech/gifapp/MainActivity.java

```diff
@@ -0,0 +1,185 @@
+    private String searchUrl;
+    private int calledMethod = 0;  // 0 -> Trending || 1 -> Search // othebe: Should use var names for this.
+    private Utility utility = new Utility();
+    String API_KEY = "dc6zaTOxFJmzC"; // Public beta key for the API // othebe: Should be final.
+                                                                        // othebe: Would have liked this to be in a config class.
+    private Integer trendingOffset = 0;
+    String trendingUrl = "http://api.giphy.com/v1/gifs/trending?api_key=" + API_KEY + "&limit=25" + "&offset=";
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        // linking to xml file
+        setContentView(R.layout.activity_main);
+
+        // linking individual UI components
+        mSearch = (EditText) findViewById(R.id.search_text);
+        mRecycleView = (RecyclerView) findViewById(R.id.recycler_view);
+        mProgressBar = (ProgressBar) findViewById(R.id.progress_bar);
+
+        // after search if the user starts scrolling the keyboard goes away automatically
+        // othebe: use separate function for listeners/handlers.
+        // othebe: could have hidden keyboard on scroll too.
+        mRecycleView.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                utility.hideKeyboard(MainActivity.this);
+                return false;
+            }
+        });
+
+        // setting the recycle view to a grid layout
+        final StaggeredGridLayoutManager sglm = new StaggeredGridLayoutManager(3, StaggeredGridLayoutManager.VERTICAL);
+        mRecycleView.setLayoutManager(sglm);
+        mRecycleView.setHasFixedSize(true);
+
+        // setting the list to the adapter
+        listOfGifs = new ArrayList<>();
+        gifAdapter = new GIFAdapter(MainActivity.this, listOfGifs);
+
+        // othebe: Small typos here and there.
+        // setting the adapter to the recyle view
+        mRecycleView.setAdapter(gifAdapter);
+
+        // calling the volley method to load trending gifs by default
+        makeGifJSONRequest(trendingUrl + trendingOffset);
+
+
+        // enabling loading additional gifs
+        mRecycleView.setOnScrollListener(new EndlessRecyleOnScrollListenerStaggered(sglm) {
+            @Override
+            public void onLoadMore(int current_page) {
+                // Check if URL is equal to trending url;
+                if (calledMethod == 1) { // for searching
+                    // othebe: offset logic should be encapsulated in a funtion.
+                    offset += 25;
+                    makeGifJSONRequest(searchUrl + offset);
+                } else {
+                    trendingOffset += 25; // for trending
+
+                    // othebe: No need to pass variables since they are part of object.
+                    makeGifJSONRequest(trendingUrl + trendingOffset);
```

Comment:
Instead of passing private properties to private methods, we should just create appropriate wrappers: makeTrendingGifJsonRequest / makeSearchGifJsonRequest around makeGifJsonRequest.